workflow/controller/operator.go

```diff
@@ -2366,6 +2385,30 @@ func (woc *wfOperationCtx) initializeNode(nodeName string, nodeType wfv1.NodeTyp
 	return &node
 }
 
+// Update a node status that has been cached and marked as finished
+func (woc *wfOperationCtx) updateAsCacheNode(node *wfv1.NodeStatus, memStat *wfv1.MemoizationStatus) *wfv1.NodeStatus {
+	node.MemoizationStatus = memStat
+
+	woc.wf.Status.Nodes[node.ID] = *node
+	woc.updated = true
+
+	return node
+}
```

Comment:
I notice this function has the same comment as the one below - is it right?

Reply:
My mistake. for `updateAsCacheNode`, I think `Update a node status with cache status` comment is proper.